pkg/api/graph/test/unpushable-build.yaml

```diff
@@ -0,0 +1,161 @@
+apiVersion: v1
+items:
+- apiVersion: v1
+  kind: BuildConfig
+  metadata:
+    creationTimestamp: null
+    labels:
+      app: ruby
+    name: ruby-hello-world
+  spec:
+    output:
+      to:
+        kind: ImageStreamTag
+        name: ruby-hello-world:latest
+    resources: {}
+    source:
+      git:
+        uri: https://github.com/openshift/ruby-hello-world
+      type: Git
+    strategy:
+      dockerStrategy:
+        from:
+          kind: ImageStreamTag
+          name: ruby-20-centos7:latest
+      type: Docker
+    triggers:
+    - github:
+        secret: LyddbeCAaw1a0x08xz9n
+      type: GitHub
+    - generic:
+        secret: ZnYJJeEvo1ri0Gk0f6YY
+      type: Generic
+    - imageChange: {}
+      type: ImageChange
+  status:
+    lastVersion: 0
+- apiVersion: v1
+  kind: Build
+  metadata:
+    creationTimestamp: null
+    labels:
+      app: ruby
+      buildconfig: ruby-hello-world
+    name: ruby-hello-world-1
+  spec:
+    output:
+      to:
+        kind: ImageStreamTag
+        name: ruby-hello-world:latest
+    resources: {}
+    serviceAccount: builder
+    source:
+      git:
+        uri: https://github.com/openshift/ruby-hello-world
+      type: Git
+    strategy:
+      dockerStrategy:
+        from:
+          kind: DockerImage
+          name: openshift/ruby-20-centos7:latest
+      type: Docker
+  status:
+    config:
+      name: ruby-hello-world
+    phase: New
+- apiVersion: v1
+  kind: ImageStream
+  metadata:
+    annotations:
+      openshift.io/image.dockerRepositoryCheck: 2015-07-06T19:05:12Z
+    creationTimestamp: null
+    labels:
+      app: ruby
+    name: ruby-20-centos7
+  spec:
+    dockerImageRepository: openshift/ruby-20-centos7
+  status:
+    dockerImageRepository: ""
```

Comment:
So this is actually ok. If you create with spec.dockerImageRepository=foo and status.dockerImageRepository="", we'll fill in status.dockerImageRepository at creation.